notificationservice/notification.go

```diff
@@ -0,0 +1,423 @@
+	return
+}
+
+func (conn *Connection) setUpSqsPolicy() (err error) {
+	jsonStringPolicy, err := conn.createJsonPolicy()
+	if err != nil {
+		println("error creating json", err.Error())
+	}
+
+	queueAttributes := sqs.SetQueueAttributesInput{
+		QueueUrl: conn.sqsQueueOutput.QueueUrl,
+		Attributes: map[string]*string{
+			sqs.QueueAttributeNamePolicy: &jsonStringPolicy,
+		},
+	}
+	conn.SqsClient.SetQueueAttributes(&queueAttributes)
+
+	conn.context.Logger.Info("Created SQS access policy for SNS Topic: " + sinkName)
+	return
+}
+
+type PolicyDocument struct {
+	Version   string
+	Statement []StatementEntry
+}
+type StatementEntry struct {
+	Sid       string
+	Effect    string
+	Action    []string
+	Resource  string
+	Condition Condition
+	Principal string
+}
+type Condition struct {
+	StringEquals map[string]string
+}
+
+func (conn *Connection) createJsonPolicy() (jsonStringPolicy string, err error) {
+
+	policy := PolicyDocument{
+		Version: "2012-10-17",
+		Statement: []StatementEntry{
+			{
+				Effect: "Allow",
+				Action: []string{
+					"SQS:*",
+				},
+				Resource: *conn.sqsQueueAttributes.Attributes[sqs.QueueAttributeNameQueueArn],
+				Condition: Condition{
+					StringEquals: map[string]string{"aws:SourceArn": *conn.TopicArn},
+				},
+				Principal: "*",
+			},
+		},
+	}
+
+	jsonPolicy, err := json.Marshal(policy)
+	jsonStringPolicy = string(jsonPolicy)
+	return
+}
+func (conn *Connection) getSnsTopicAttributes() (topicExists bool, err error) {
+	conn.snsClient = sns.New(conn.session)
+
+	topicExists = false
+	listTopicsInput := sns.ListTopicsInput{}
+	err = conn.snsClient.ListTopicsPages(&listTopicsInput,
+		func(output *sns.ListTopicsOutput, lastPage bool) bool {
+			for topicNum := range output.Topics {
+				if strings.Contains(*output.Topics[topicNum].TopicArn, sinkName) {
+					topicExists = true
+					conn.TopicArn = output.Topics[topicNum].TopicArn
+					return false
+				}
+			}
+			return true
+		})
+	return
+}
+func (conn *Connection) getSqsQueueAttributes() (queueExists bool, err error) {
+	conn.SqsClient = sqs.New(conn.session)
+
+	queueExists = false
+	listQueuesInput := sqs.ListQueuesInput{
+		QueueNamePrefix: &sinkName,
+	}
+
+	listQueuesOutput, err := conn.SqsClient.ListQueues(&listQueuesInput)
+
+	for queueNum := range listQueuesOutput.QueueUrls {
+		if strings.Contains(*listQueuesOutput.QueueUrls[queueNum], sinkName) {
+			queueExists = true
+			conn.sqsQueueOutput = &sqs.CreateQueueOutput{
+				QueueUrl: listQueuesOutput.QueueUrls[queueNum],
+			}
+
+			arnAttribute := sqs.QueueAttributeNameQueueArn
+			queueAttributesInput := sqs.GetQueueAttributesInput{
+				AttributeNames: []*string{&arnAttribute},
+				QueueUrl:       conn.sqsQueueOutput.QueueUrl,
+			}
+			conn.sqsQueueAttributes, err = conn.SqsClient.GetQueueAttributes(&queueAttributesInput)
+			if err != nil {
+				return
+			}
+			return
+		}
+	}
+	return
+}
+
+func (conn *Connection) deleteSnsTopic() (err error) {
```

Comment:
I don't see deleting a policy here - is it taken care when deleting SQS queue?

Reply:
Those are parameters of the SQS queue, the policy itself is not created - only set up as an attribute of the SQS queue.